### PR TITLE
Feature/legal templates

### PR DESCRIPTION
**Pull request by noctera**
|               Status                |                Type                 | Env Vars Change |
| :---------------------------------: | :---------------------------------: | :-------------: |
| :white_check_mark: Ready | Feature |     Yes      |

## Description

This PR adds the function to create your own routes and add static pages or redirects to your server in them. Mainly to include a privacy policy and terms and conditions.

just define your custom pages in the config (documentation will be done soon)

```js
// ...
pages: {
    privacy: {
      url: '/privacy',
      fallback: { type: 'file', location: './staticPages/privacy-en.html' },
      langs: {
        en: { type: 'redirect', location: 'https://my-website/privacy' },
        de: { type: 'file', location: './staticPages/privacy-de.html' },
        // ...
      },
    },
   termsAndConditions: {
      url: '/terms-and-conditions',
      fallback: { type: 'file', location: './staticPages/privacy-en.html' },
      langs: {
        en: { type: 'redirect', location: 'https://my-website/privacy' },
        // ...
      },
    },
  },
```

Info: The ``langs`` prop is not required.

In this case ``/privacy`` and ``/terms-and-conditions`` would return or redirect the fallback page, as no language is given in the query params. If you add ``?lang=en`` the page, defined in ``langs`` will be displayed. 

To Do:
- [x] Joi schema

If you use a language as query parameter, that is not defined in the config, you will be redirected to the fallback page.

## Motivation and Context

We will need this function to bring the Vocascan Cloud one step closer to the public

## Checklist

- [x] I have read the **CONTRIBUTING** document.
- [x] I have considered the accessibility of my changes (i.e. did I add proper content descriptions to images, or run my changes with talkback enabled?)
- [x] I have documented my code if needed

### CODE REVIEW

File: app/Controllers/StaticPagesController.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const catchAsync = require('../utils/catchAsync');
+const path = require('path');
+const ApiError = require('../utils/ApiError.js');
+const httpStatus = require('http-status');
+
+const renderHandler = catchAsync(async (res, page) => {
+  const options = { root: path.join(__dirname, '../../') };
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js]: what about making it relative to `process.cwd()` ? I used that a while ago also for the db path. So the files/folders are relative to the current folder the `vocascan-server web` cmd was executed.

File: app/Controllers/StaticPagesController.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  // check if page should be rendered from a static file or should be redirected to another one
+  if (page.type === 'file') {
+    res.sendFile(page.location, options, (err) => {
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js:10]: shouldnt we use a template engine here? Or is that something for later PRs? By using a template engine, we are able to let the template engine decide which page should be included.

File: app/Controllers/StaticPagesController.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  // check which language the user wants his page
+  const language = req.query.lang;
+
+  // if no language is given, render fallback page
+  if (!language) {
+    const fallbackPage = pageValues.fallback;
+
+    renderHandler(res, fallbackPage);
+
+    return;
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js]: we can simplify that
```suggestion
    return renderHandler(res, pageValues.fallback);
```

File: routes/pages.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+
+const config = require('../app/config/config');
+
+const StaticPagesController = require('../app/Controllers/StaticPagesController.js');
+
+const router = express.Router();
+
+if (config.pages) {
+  for (const [, value] of Object.entries(config.pages)) {
```
Comment (luwol03) [inline routes/pages.js]: Use `Object.values(...)` if you dont need the keys.

File: app/Controllers/StaticPagesController.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      if (err) {
+        throw new ApiError(httpStatus.NOT_FOUND, 'Page was not found');
+      }
+    });
+  } else {
+    res.redirect(page.location);
+  }
+});
+
+const renderPage = catchAsync(async (req, res, pageValues) => {
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js]: I meant on discord (see at the bottom for the second part):
```suggestion
const getRenderPageHandler = (pageValues) => catchAsync(async (req, res) => {
```

File: routes/pages.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    router.get(value.url, (req, res) => {
+      StaticPagesController.renderPage(req, res, value);
+    });
```
Comment (luwol03) [inline routes/pages.js]: Second part:
```suggestion
    router.get(value.url, StaticPagesController.getRenderPageHandler(value));
```

File: app/Controllers/StaticPagesController.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+const catchAsync = require('../utils/catchAsync');
+const path = require('path');
+const ApiError = require('../utils/ApiError.js');
+const httpStatus = require('http-status');
+
+const renderHandler = catchAsync(async (res, page) => {
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js]: then this sould be `renderPageHandler` to be more specific.

**Review by luwol03 — CHANGES_REQUESTED** [review]

File: app/Controllers/StaticPagesController.js (hunk 2 above)
Comment (noctera) [reply app/Controllers/StaticPagesController.js:10]: I think this is more like a task for a later PR. This if query is just to determine whether we should use ``sendFile`` or ``redirect``

File: app/Controllers/StaticPagesController.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  const options = { root: process.cwd() };
+
+  // check if page should be rendered from a static file or should be redirected to another one
+  if (page.type === 'file') {
+    res.sendFile(page.location, options, (err) => {
+      if (err) {
+        throw new ApiError(httpStatus.NOT_FOUND, 'Page was not found');
+      }
+    });
+  } else {
```
Comment (luwol03) [inline app/Controllers/StaticPagesController.js]: please check if type is redirect. This is much cleaner. If you want, you can add a default to the schema.

**Review by luwol03 — CHANGES_REQUESTED** [review]

**Review by luwol03 — APPROVED** [review]

**Pull request merged**